Userland/Libraries/LibSQL/Heap.cpp

```diff
@@ -33,30 +46,39 @@ Heap::Heap(String file_name)
 
     auto file_or_error = Core::File::open(name(), Core::OpenMode::ReadWrite);
     if (file_or_error.is_error()) {
-        warnln("Couldn't open '{}': {}", name(), file_or_error.error());
-        VERIFY_NOT_REACHED();
+        return Error::from_string_literal(String::formatted("Heap::open({}): could not open: {}", name(), file_or_error.error()));
     }
     m_file = file_or_error.value();
-    if (file_size > 0)
-        read_zero_block();
-    else
+    if (file_size > 0) {
+        if (auto error_maybe = read_zero_block(); error_maybe.is_error()) {
+            m_file = nullptr;
+            return error_maybe.error();
+        }
+    } else {
         initialize_zero_block();
-    dbgln_if(SQL_DEBUG, "Heap file {} opened. Size = {}", file_name, size());
+    }
+    dbgln_if(SQL_DEBUG, "Heap file {} opened. Size = {}", name(), size());
+    return {};
 }
 
 ErrorOr<ByteBuffer> Heap::read_block(u32 block)
 {
+    if (m_file.is_null())
+        return Error::from_string_literal(String::formatted("Heap({})::read_block({}): Heap file not opened"sv, name(), block));
+
     auto buffer_or_empty = m_write_ahead_log.get(block);
     if (buffer_or_empty.has_value())
         return buffer_or_empty.value();
 
-    VERIFY(block < m_next_block);
+    if (block >= m_next_block)
+        return Error::from_string_literal(String::formatted("Heap({})::read_block({}): block # out of range (>= {})"sv, name(), block, m_next_block));
+
     dbgln_if(SQL_DEBUG, "Read heap block {}", block);
-    if (!seek_block(block))
-        VERIFY_NOT_REACHED();
+    if (auto error_maybe = seek_block(block); error_maybe.is_error())
+        return Error(error_maybe.error());
```

Comment:
I'm less sure about this one because of the explicit wrapping you needed, but maybe this will work?
```suggestion
    TRY(seek_block(block));
```